src/utils/objectUtils.ts

```diff
@@ -0,0 +1,78 @@
+export function copyWithoutUndefined<T extends Record<K, V>, K extends string | number | symbol, V>(
+  originalValue: T,
+): T {
+  return Object.keys(originalValue).reduce((acc, key) => {
+    // @ts-ignore
+    if (originalValue[key] !== undefined) {
+      // @ts-ignore
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      acc[key] = originalValue[key]
+    }
+    return acc
+  }, {} as Record<string, unknown>) as T
+}
```

Comment:
This can be expressed more neatly:
```ts
export function copyWithoutUndefined<T extends Record<K, V>, K extends PropertyKey, V>(
  originalValue: T,
): T {
  return Object.fromEntries(Object.entries(originalValue).filter(([_key, value]) => value !== undefined)) as T;
}
```